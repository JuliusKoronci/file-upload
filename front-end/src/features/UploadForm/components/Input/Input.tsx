import { Button, Grid, Input as MUIInput } from '@material-ui/core';
import { Save } from '@material-ui/icons';
import { SFC } from 'react';
import * as React from 'react';
import { withState } from 'recompose';
import { styles } from '../../../../styles';
import { uploadService } from '../../../../domains/Documents';

export interface IInputProps {
  selectedFile: any;
  handleInputChange: any;
  handleUpload: any;
}

/**
 * Simple Upload Input field component
 *
 * @param {File} selectedFile - selected file from the input
 * @param {Function} handleInputChange - update state with selected file
 */
const Input: SFC<IInputProps> = ({ selectedFile, handleInputChange, handleUpload }) => {
  const handleChange = (event: any) => handleInputChange(event.target.files[0]);
  const handleButtonClick = () => uploadService(selectedFile);
  
  return (<Grid container={true} spacing={24}>
    <Grid item={true} xs={12}>
      <MUIInput
        onChange={handleChange}
        style={styles.uploadForm} type="file"
      />
    </Grid>
    <Grid item={true} xs={12}>
      <Button
        disabled={!selectedFile}
        variant="contained"
        color="primary"
        onClick={handleButtonClick}
      >
        <Save />&nbsp;
        Upload
      </Button>
    </Grid>
  </Grid>);
};

export default withState('selectedFile', 'handleInputChange', null)(Input);