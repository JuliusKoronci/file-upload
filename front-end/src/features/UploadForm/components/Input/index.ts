export { default as Input, IInputProps } from './Input';