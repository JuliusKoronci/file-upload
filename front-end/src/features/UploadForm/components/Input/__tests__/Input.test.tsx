import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../Input';


describe('Message', () => {
  it('should render snapshot', () => {
    expect(toJson(shallow(<Input
      handleUpload={jest.fn()}
      loading={false}
    />))).toMatchSnapshot();
  });
});