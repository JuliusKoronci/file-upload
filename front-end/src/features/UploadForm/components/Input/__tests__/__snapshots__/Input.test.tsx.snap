// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render snapshot 1`] = `
<Input
  handleInputChange={[Function]}
  handleUpload={[Function]}
  loading={false}
  selectedFile={null}
/>
`;
